### PR TITLE
#83 Implemented comment REST endpoint to handle POST requests.

### PR DESCRIPTION
**Pull request by AhmadHashems**
Created the comment REST endpoint to handle POST requests, and added the test cases for the endpoint as well.

### CODE REVIEW

**Review by OliverA100 — CHANGES_REQUESTED** [review]
By wrapping the API endpoint with permission_classes([IsAuthenticated]) some post requests might be blocked from accessing this endpoint. If there is no specific reason to keep this line please remove it.

File: api/views.py [hunk 1]
```diff
@@ -47,6 +46,20 @@ def api_post(request):
         return Response({"errors": error_messages})
 
 
+@api_view(['POST'])
+@authentication_classes([SessionAuthentication, BasicAuthentication])
+@permission_classes([IsAuthenticated])
```
Comment (OliverA100) [inline api/views.py:51]: Please review this line, as mentioned in a previous "review changes" comment.

Comment (AhmadHashems) [reply api/views.py:51]: @permission_classes([IsAuthenticated]) is needed as if the user is not logged in, they cannot write a new comment. So, this allows only the authenticated users only to post a comment.

Comment (OliverA100) [reply api/views.py:51]: Perfect thanks for the explanation.

**Review by OliverA100 — APPROVED** [review]

**Review by OliverA100 — APPROVED** [review]
All looks good! Will now merge the pull request to main.

**Pull request merged**